Add unix socket

matthiasbeyer opened the pull request:
This is a competing PR to #466 to *_finally_ get this stuff done.

matthiasbeyer committed:
Add the ws_socket type

matthiasbeyer committed:
Add doxygen comment frame

matthiasbeyer committed:
Add ws_socket_init()'s definition

matthiasbeyer committed:
Add socket_build_connection_cb()'s implementation

matthiasbeyer committed:
Add ws_socket_init()'s implementation

matthiasbeyer commented:
Please review and tell me what to do.

matthiasbeyer commented:
And yes, I know it doesn't build... I don't know how to integrate it, yet.

neithernut commented on src/util/socket.c:
```diff
@@ -36,10 +37,102 @@
+    struct ev_io* io,
+    int revents
+) {
+    struct ws_socket* s = (struct ws_socket*) io;
```
Use the data field in io.

neithernut commented on src/util/socket.c:
```diff
@@ -25,21 +25,86 @@
 #include "util/socket.h"
+#include "serialize/json/deserializer.h"
+#include "serialize/json/serializer.h"
+#include "connection/processor.h"
```
Remove these includes.

neithernut commented:
ACK for squashing.

matthiasbeyer commented:
Please review carefully. I squashed almost everything and rewrote the history completely, as you can see.

Hopefully there are no rebase-conflicts left...

neithernut commented:
ACK.
